src/main/java/org/jabref/logic/util/io/FileUtil.java

```diff
@@ -286,11 +286,13 @@ public static String createDirNameFromPattern(BibDatabase database, BibEntry ent
             targetName = entry.getCiteKeyOptional().orElse("default");
         }
 
-        // Removes illegal characters from filename
+        // Removes illegal characters from directory name
         targetName = FileNameCleaner.cleanDirectoryName(targetName);
 
-        // Truncates long directory names
-        targetName = getValidFileName(targetName);
+        // Only use valid filenames for the directories in the path
+        targetName = Arrays.stream(targetName.split(Pattern.quote(File.separator)))
+                            .map(FileUtil::getValidFileName)
+                            .collect(Collectors.joining(File.separator));
```

Comment:
Note that on my system the maximum **file**name and **directory** name is 255 characters while the absolute **path**name appears to be limited to 1024 characters on FAT32.
Windows [appear to have a maximum path length of 260](https://docs.microsoft.com/en-gb/windows/win32/fileio/naming-a-file#maximum-path-length-limitation) which I'd consider an argument for not including these lines.

Reply:
Java will throw an IO exception error if it's not possible, so we should not modify the (generated directory structure), just the filename.  Your bib file or file directory could also be on an network drive and that length limit does not apply to those,... 
=>So remove these lines.

Reply:
Won't it throw the same exception for an invalid filename? -> remove the check for filename length?

Reply:
Sure, but filenames are a bit different, as they are usually auto generated from a download. If you have a paper with a long title and some other maybe illegal characters, you don't want to fix it manually. That's why we only care for filenames.

Reply:
Ok. The lines are removed! As always, thank you for taking the time to look through and answer questions!